codebase-pipelines/custom_pipeline_update/test_update_pipeline.py

```diff
@@ -0,0 +1,50 @@
+import pytest
+import unittest
+from unittest.mock import MagicMock, patch
+
+from update_pipeline import update_pipeline_stage_failure
+
+
+class TestUpdatePipeline(unittest.TestCase):
+
+    def test_update_pipeline_stage_failure_sets_rollback(self):
+        with patch("boto3.client") as mock_boto_client:
```

Comment:
To bring this in line with the way we are structuring tests elsewhere, could you move  the boto client instantiation into the `if __name__ == "__main__":` block and pass it into `update_pipeline_stage_failure(client, ["test_pipeline")`

Then you can remove the patching and just pass in a mock.